src/services/expenses.service.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const Expenses = require('../models/expenses.model');
+
+const getAllExpenses = async(userId, categories, from, to) => {
+  let expenses = await Expenses.findAll();
+
+  if (categories && userId) {
+    expenses = expenses.filter(item => item.category === categories);
```

Comment:
It appears that we are implementing filtering logic on the server rather than allowing the database to handle it for us. It is generally more efficient to retrieve from the database only the values that are actually needed.

You could approach it like that:
```javascript
const whereClause = {};

if(userId) {
  whereClause.userId = userId
}

// and so on...
```

and then, at the very end
```javascript
Expense.findAll({ where: whereClause });
```